include/zephyr/sw_isr_table.h

```diff
@@ -22,6 +22,15 @@
 extern "C" {
 #endif
 
+/* When dynamic interrupts are disabled, mark IRQ vector tables as const.
+ * This is required to prevent linker from marking section as writable.
+ */
+#ifdef CONFIG_DYNAMIC_INTERRUPTS
+#define ISR_CONST_TABLE
```

Comment:
I'd say we shouls rename `ISR_CONST_TABLE` as it might not be const so this is misleading.

Probably `ISR_TABLE_ATTR` or something like that?